src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -852,7 +854,10 @@ bool ClassLoader::add_to_app_classpath_entries(JavaThread* current,
   }
 
   if (entry->is_jar_file()) {
-    ClassLoaderExt::process_jar_manifest(current, entry);
+    bool is_multi_release = ClassLoaderExt::process_jar_manifest(current, entry);
+    if (is_multi_release) {
+      entry->set_multi_release_jar();
```

Comment:
Maybe the set_multi_release_jar() call should be moved into process_jar_manifest()? That way, the logic doesn't need to be split in two places.

Reply:
That's a good idea. I've pushed an update.